osf/migrations/0161_add_registration_files_count.py

```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.15 on 2019-01-18 14:43
+from __future__ import unicode_literals
+import logging
+import progressbar
+
+from django.db import migrations, models
+from django.db.models import Count
+from django_bulk_update.helper import bulk_update
+from osf.models import Registration
+
+logger = logging.getLogger(__file__)
+
+def add_registration_files_count(state, *args, **kwargs):
+    registrations = Registration.objects.filter(is_deleted=False).filter(
+        files__type='osf.osfstoragefile',
+        files__deleted_on__isnull=True
+    ).annotate(
+        annotated_file_count=Count('files')
+    )
+    progress_bar = progressbar.ProgressBar(maxval=registrations.count() or 100).start()
+    registrations_to_update = []
+
+    for i, registration in enumerate(registrations, 1):
+        progress_bar.update(i)
+        registration.files_count = registration.annotated_file_count
+        registrations_to_update.append(registration)
+
+    bulk_update(registrations_to_update, update_fields=['files_count'], batch_size=5000)
+    logger.info('Populated `files_count` on a total of {} registrations'.format(len(registrations_to_update)))
+    progress_bar.finish()
```

Comment:
Added a progress bar, but this part flies, the majority of the time is now in annotating the query, and then bulk_updating at the end, not the setting of the field in the `for` loop.